src/components/BaseTextArea/BaseTextArea.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <textarea
+    ref="textareaRef"
+    :autocomplete="autocomplete"
+    :rows="rows"
+    :value="value"
+    v-on="inputListeners"
+  />
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  onMounted,
+  ref,
+} from '@vue/composition-api';
+import autosize from 'autosize';
+import { Nullable } from '@tager/admin-services';
+
+type ResizePropType = 'none' | 'both' | 'horizontal' | 'vertical';
+type AutocompletePropType = 'on' | 'of';
+
+interface Props {
+  value: string;
+  resize: ResizePropType;
+  autosize: boolean;
+  rows: number;
+  autocomplete: AutocompletePropType;
+}
+
+export default defineComponent<Props>({
+  name: 'BaseTextArea',
+  props: {
+    value: {
+      type: String,
+      default: '',
+    },
+    autosize: {
+      type: Boolean,
+      default: true,
+    },
+    rows: {
+      type: Number,
+      default: 3,
+    },
+    autocomplete: {
+      type: String,
+      default: 'off',
+      validator(value: string) {
+        return ['on', 'off'].includes(value);
+      },
+    },
+  },
+  setup(props, context) {
+    const textareaRef = ref<Nullable<HTMLTextAreaElement>>(null);
+
+    const inputListeners = computed(() => ({
+      ...context.listeners,
+      input: (event: Event) => {
+        const { value } = event.target as HTMLInputElement;
+        context.emit('input', value);
+      },
+      change: (event: Event) => {
+        const { value } = event.target as HTMLInputElement;
+        context.emit('change', value);
+      },
+    }));
+
+    onMounted(() => {
+      if (props.autosize && textareaRef.value) {
+        autosize(textareaRef.value);
+      }
+    });
+
```

Comment:
run `autosize.destroy` in `onUnmounted` hook

https://composition-api.vuejs.org/api.html#lifecycle-hooks
https://github.com/jackmoore/autosize/blob/master/src/autosize.js#L242